### PR TITLE
feat: use new Sourcegraph extension API

### PR DESCRIPTION
**Pull request by sqs**
Uses the new extension API from https://github.com/sourcegraph/sourcegraph-extension-api/pull/56

Depends on https://github.com/sourcegraph/extensions-client-common/pull/36

### CODE REVIEW

File: src/libs/github/inject.tsx [hunk 1]
```diff
@@ -133,10 +133,10 @@ function injectCodeIntelligence(): void {
     if (isSingleCodeFile && useExtensions && filePath) {
         extensionsContextController = createExtensionsContextController(sourcegraphUrl)
         extensionsController = createController(extensionsContextController.context, createMessageTransports)
-        simpleProviderFns = createLSPFromExtensions(extensionsController)
+        simpleProviderFns = createLSPFromExtensions(extensionsController!)
 
-        const constExtensionsContextController = extensionsContextController
-        const constController = extensionsController
+        const constExtensionsContextController = extensionsContextController!
+        const constController = extensionsController!
```
Comment (chrismwendt) [inline src/libs/github/inject.tsx]: This typechecks without `!` for me locally. Did it not for you?

File: src/extension/scripts/background.tsx [hunk 2]
```diff
@@ -393,7 +395,18 @@ function spawnWebWorkerFromURL(url: string): Promise<Worker> {
...
+            try {
+                const worker = new ExtensionHostWorker()
+                const initData: InitData = { bundleURL: blobURL, sourcegraphURL: sourcegraphUrl }
+                worker.postMessage(initData)
+                return worker
+            } catch (err) {
+                console.error(err)
+            }
+            throw new Error('failed to initialize extension host')
+        })
```
[the remark this answers is not in the document]
Comment (sqs) [reply src/extension/scripts/background.tsx]: I think it can be moved to ecc

File: src/libs/github/inject.tsx (hunk 1 above)
Comment (sqs) [reply src/libs/github/inject.tsx]: It didn’t but that might be me running the wrong TS version in multiroot vscode. Will check

**Comment by chrismwendt** [comment]
@sqs I merged #161, so you'll need to rebase this branch (hopefully there aren't too many conflicts).

**Pull request merged**